template/src/main/resources/application.yml

```diff
@@ -29,16 +30,21 @@ amqp:
     {% endif %}
   {% endfor %}
 {% endif %}
+
 {% if server.protocol() == 'mqtt' %}
 mqtt:
   broker: {% for line in server.description() | splitByLines %}
     # {{line | safe}}{% endfor %}
     address: {% if server.variable('port') %}{{server.url() | replace('{port}', server.variable('port').defaultValue())}}{% else %}{{server.url()}}{% endif %}
     username:
     password:
-    {% if server.binding('mqtt') and server.binding('mqtt').clientId %}clientId: {{server.binding('mqtt').clientId}}{% endif %}
-    {% if server.binding('mqtt') and server.binding('mqtt').cleanSession | isDefined %}cleanSession: {{server.binding('mqtt').cleanSession}}{% endif %}
-    {% if server.binding('mqtt') and server.binding('mqtt').lastWill %}lastWill:
+    {% if server.binding('mqtt') and server.binding('mqtt').clientId and (params.mqttClientId == 'guest') %}
```

Comment:
I see the problem if in AsyncAPI `server.binding('mqtt').clientId` = `provider` (example), but user would like to intentionally change it to `guest` via parameter.

Reply:
I see. I had an approach where I made use of an if-elif condition like this - 

```
{% if params.mqttClientId %}
clientId : {{ params.mqttClientId }}
{% elif if server.binding('mqtt') and server.binding('mqtt').clientId %}
clientId: {{ server.binding('mqtt').clientId }}
{% else %}
clientId: defaultValue
{% endif %}
```

But this requires the default value to be removed from `package.json` and I wasn't sure if it was better. What do you think?